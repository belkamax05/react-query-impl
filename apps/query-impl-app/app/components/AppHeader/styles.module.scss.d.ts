export type Styles = {
  links: string;
  root: string;
};

export type ClassNames = keyof Styles;

declare const styles: Styles;

export default styles;
