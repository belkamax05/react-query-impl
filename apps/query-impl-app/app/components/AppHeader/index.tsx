import Link from 'next/link';
import styles from './styles.module.scss';

const AppHeader = () => {
  return (
    <header className={styles['root']}>
      <div className={styles['links']}>
        <Link href="/">Home</Link>
        <Link href="/samples">Samples</Link>
      </div>
    </header>
  );
};

export default AppHeader;
