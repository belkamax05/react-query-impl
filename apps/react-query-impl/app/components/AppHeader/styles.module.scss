.root {
  padding: 16px 8px;
  border-bottom: 1px solid #484746;
}
.links {
  display: flex;
  gap: 8px;
}
