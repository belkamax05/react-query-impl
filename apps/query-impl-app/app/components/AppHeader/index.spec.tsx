import { render } from '@testing-library/react';
import AppHeader from '.';

describe('AppHeader', () => {
  it('should render the header with the correct links', async () => {
    const { baseElement, getByText } = render(<AppHeader />);
    const homeLink = getByText('Home');
    const samplesLink = getByText('Samples');

    expect(homeLink).toBeInTheDocument();
    expect(samplesLink).toBeInTheDocument();
    expect(baseElement).toMatchSnapshot();
  });
});
