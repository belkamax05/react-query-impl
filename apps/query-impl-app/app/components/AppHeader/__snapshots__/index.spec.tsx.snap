// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppHeader should render the header with the correct links 1`] = `
<body>
  <div>
    <header
      class="root"
    >
      <div
        class="links"
      >
        <a
          href="/"
        >
          Home
        </a>
        <a
          href="/samples"
        >
          Samples
        </a>
      </div>
    </header>
  </div>
</body>
`;
